### PR TITLE
Update deploy instructions for Docker

### PR DESCRIPTION
**Pull request by azigler**
#### Summary
This PR is part of a larger project to improve the deploy flow for Docker. It updates [Local machine setup using Docker](https://docs.mattermost.com/install/setting-up-local-machine-using-docker.html) to have instructions for both Preview Mode and production, and distills those instructions into their simplest forms.

### CODE REVIEW

**Comment by azigler** [comment]
I improved the navigation links to take advantage of the Docker install page's new anchor headings. We're now more clearly directing users to the preview vs. production versions of Docker wherever appropriate. When this is merged, we can close #5509.

**Review by BenLloydPearson — APPROVED** [review]
I made some slight tweaks to the Docker page, renamed the file, created a redirect, and updated the LHS navigation. This all looks great to me now except for the fact that the redirect doesn't appear to work on the preview. @cwarnermm is this expected behavior?

File: source/guides/deployment.rst
Comment (cwarnermm) [inline source/guides/deployment.rst:23]:
```suggestion
* :doc:`Docker deployment </install/install-docker>` - Preview Mattermost instantly or deploy via Docker for production use.
```

File: source/install/install-docker.rst
Comment (cwarnermm) [inline source/install/install-docker.rst:95]:
```suggestion
Deploy Mattermost on Docker For production use
```

File: source/install/install-docker.rst
Comment (cwarnermm) [inline source/install/install-docker.rst:95]:
```suggestion
Deploy Mattermost on Docker for production use
```

File: source/install/install-docker.rst
Comment (cwarnermm) [inline source/install/install-docker.rst:163]:
```suggestion
5. Configure SSO with GitLab *(optional)*:
```

File: source/install/install-docker.rst
Comment (cwarnermm) [inline source/install/install-docker.rst:175]:
```suggestion
6. Deploy
```

**Comment by cwarnermm** [comment]
@BenLloydPearson - Yes, it's expected behaviour in cases where you're creating a brand new page to redirect to that isn't yet available on production. 

I've made small tweaks to case (as we're moving to sentence case across the full doc set), and I updated procedural numbering to remove duplicates.

**Review by cwarnermm — APPROVED** [review]
This is a great update, @azigler! Thank you!

**Review by nab-77 — APPROVED** [review]
Thanks for this @azigler! this is a most welcome update (and fantastic to bring from GitHub README.md into docs.mm.com. That said, I think this contains more steps than needed for users (e.g. the checkout for ENV files).

Let's get this great update out and in parallel I will work with the release team to see where changes to make setup even easier.

**Comment by azigler** [comment]
I noticed that the GitLab instructions were missing a second portion, so I updated them to reflect the full step.

**Comment by azigler** [comment]
Once this PR is merged, [mattermost-server/PR#19924](https://github.com/mattermost/mattermost-server/pull/19924) and [mattermost-webapp/PR#10137](https://github.com/mattermost/mattermost-webapp/pull/10137) should both be merged to reflect the new Docker deploy instructions link.

**Comment by BenLloydPearson** [comment]
@cwarnermm I think this PR can be merged now. It is blocking a PR to the [Docker readme](https://github.com/mattermost/docker/pull/92) and we're expecting to roll out our website changes next week, so we'll need this page to be live.

File: source/guides/deployment.rst
Comment (cwarnermm) [inline source/guides/deployment.rst:83]:
```suggestion
    Install Mattermost via Docker </install/install-docker>
```

**Pull request merged**
